src/storage/src/source/generator.rs

```diff
@@ -76,14 +78,20 @@ impl SourceRender for LoadGeneratorSourceConnection {
         scope: &mut G,
         config: RawSourceCreationConfig,
         resume_uppers: impl futures::Stream<Item = Antichain<MzOffset>> + 'static,
-        _start_signal: impl std::future::Future<Output = ()> + 'static,
+        start_signal: impl std::future::Future<Output = ()> + 'static,
     ) -> (
         Collection<G, (usize, Result<SourceMessage, SourceReaderError>), Diff>,
         Option<Stream<G, Infallible>>,
         Stream<G, HealthStatusMessage>,
         Stream<G, ProgressStatisticsUpdate>,
         Vec<PressOnDropButton>,
     ) {
+        // Currently `KEY VALUE` loadgen renders its own operator, and is not a single-worker
+        // basic generator like the rest.
+        if let LoadGenerator::KeyValue(kv) = self.load_generator.clone() {
```

Comment:
I think we need an enum of `GeneratorKind { Simple(..), KeyValue(..) }` where each of the inner types has a render method.